resources/config/io.neonbee.internal.verticle.ServerVerticle.example.yaml

```diff
@@ -36,32 +36,46 @@ config:
 
     # specific endpoint configuration, defaults to the object seen below
     endpoints:
-        # provides a OData V4 compliant endpoint, for accessing entity verticle data
-        odata:
-            # enable the OData endpoint, defaults to true
-            enabled: true
-            # the base path to map this endpoint to, defaults to /odata/
-            basePath: /odata/
-            # namespace and service name URI mapping (strict, or loose based on CDS)
-            uriConversion: strict
+      # provides a OData V4 compliant endpoint, for accessing entity verticle data
+      - type: io.neonbee.endpoint.ODataV4Endpoint
+        # enable the OData endpoint, defaults to true
+        enabled: true
+        # the base path to map this endpoint to, defaults to /odata/
+        basePath: /odata/
+        # endpoint specific authentication chain, defaults to null and using the default authentication chain
+        authenticationChain: ~
+        # namespace and service name URI mapping (strict, or loose based on CDS)
+        uriConversion: strict
```

Comment:
This is wrong, isn't it? It should be:

additionalConfig:
  uriConversion: strict

Reply:
Hey @pk-work, I wanted to not overcomplicate the config file and I found it strage that you have a "additonal**Config**" in a "**config**" file. Same as I found it strage to name it "endpoints**Config**". Thus I decided to introduce new "collectAdditionalConfig" and "rephraseConfigNames" methods. However, due to that I found this issue when I fixed the unit tests, it's part of my "fix unit tests" commit and you will find it there. This change allows you to specify addtional configurations on the same level and it also allows you to simply specify "endpoints" and not "endpointConfigs" in the YAML file.

Reply:
I agree to and understood the introduction of "collectAdditionalConfig". But maybe there is a better solution instead of introducing "rephraseConfigNames". We could simply rename the variables in the DataObject or define a custom "jsonPropertyNameFormatter" in the DataObject annotation, which goes in the same direction as the "rephraseConfigNames" method.

I would prefer to evaluate "jsonPropertyNameFormatter", because then we have clean variable names and use Vert.x tools instead of inventing our own rephrasing.

Reply:
Agree that this would be the better solution, but such a thing we wold need to suggest to the Vert.x CodeGen first isn't it? Which could take a lot of time. I think my solution is not very unnice, however I agree, having it in the CodeGen would be nicer. I would suggest we'd stick with my suggestion for now and then propose it to the CodeGen project as a contribution, what do you think? I think the jsonPropertyNameFormatter is an experimental feature as far as I can judge the code... I think it'll be an easy improvement as a subsequent commit, which doesn't brake anything, thus I would suggest, we go with it for now and reuse the reprhaseConfigNames later on to put it into a jsonPropertyNameFormatter.

Renaming the fields in the DataObject is no good solution either... as we would need to name the property "endpoints" and have a "getEndpoints" and "setEndpoints" method. However the method names would be misleading, as the methods would return "EndpointConfig"s and NOT "Endpoint"s. That is why the rephraseConfigNames method thought makes a lot of sense.

Reply:
Okay I talked with Julien and it is not possible at the moment to use jsonPropertyNameFormatter for this, as the formatters you can define must be one of these [1]. But Julien likes the idea and it is appreciated to contribute this feature. 

Then let's continue as you suggested.

[1] https://github.com/vert-x3/vertx-codegen/blob/730d4f707ef4eb8d6dca489614b9a5e9e0e3c182/src/main/java/io/vertx/codegen/generators/dataobjecthelper/DataObjectHelperGen.java#L349